### PR TITLE
✨ New version without Fela and with Stitches

### PR DESCRIPTION
**Pull request by baierjak**
### Added

- ✨ Stitches integration
- ✨ Vercel CI integration 
- ✨ Generate icons script
- ✨ UI Components unit tests
- ✨ Storybook snapshot testing via Chromatic

### Changed

- ♻️ Custom gitlab pipelines 
- ♻️ Next.js and react versions
- ♻️ Storybook components moved to UI module

### Removed

- 🔥 Custom Gitlab CI settings
- 🔥 Fela styles and integrations

### CODE REVIEW

File: template/next.config.js [hunk 1]
```diff
@@ -6,6 +6,10 @@ const nextConfig = {
     reactStrictMode: true,
     swcMinify: true,
 
+    images: {
+        domains: ['localhost', 'storage.googleapis.com'],
```
Comment (adamtretera) [inline template/next.config.js]: project dependent

File: template/.svgrrc.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+module.exports = {
+    replaceAttrValues: {
+        '#828282': 'currentColor',
+        '#000': 'currentColor',
+        '#FFD618': 'currentColor',
+        '#FFD824': 'currentColor',
```
Comment (adamtretera) [inline template/.svgrrc.js]: Remove - they are project specific - keep just one (or add comment)

File: template/package.json [hunk 3]
```diff
@@ -19,29 +20,36 @@
...
-        "build-storybook": "build-storybook"
+        "build-storybook": "build-storybook",
+        "generate-icons": "svgr src/modules/ui/components/Icons/Icons/assets"
     },
     "dependencies": {
+        "@radix-ui/react-collapsible": "^1.0.1",
+        "@radix-ui/react-dialog": "^1.0.2",
+        "@radix-ui/react-icons": "^1.1.1",
+        "@radix-ui/react-navigation-menu": "^1.1.1",
+        "@radix-ui/react-select": "1.2.0",
```
Comment (adamtretera) [inline template/package.json]: Remove again specific to the project - I would just keep the Iocons if its used in the general Icon component

File: template/package.json [hunk 4]
```diff
@@ -19,29 +20,36 @@
...
-        "fela-plugin-prefixer": "^12.2.0",
-        "fela-plugin-validator": "^12.2.0",
-        "fela-tools": "^12.2.0",
+        "@stitches/react": "^1.2.8",
+        "@svgr/cli": "^6.5.1",
+        "@tanstack/react-query": "^4.24.4",
+        "@tanstack/react-query-devtools": "^4.24.4",
+        "@types/facebook-pixel": "^0.0.24",
+        "@vercel/analytics": "^0.1.11",
+        "axios": "^1.3.2",
```
Comment (adamtretera) [inline template/package.json]: Remove axios, facebook pixel,date-fns i would also remove the tanstack query and jotai (maybe add them as optinal in cli) or just add the to readme as recomended

File: template/src/modules/ui/components/Icons/Icon/Icon.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import * as Styled from './Icon.styles';
+
+export interface IconProps extends VariantProps<typeof Styled.IconBox> {
+    icon: FC<{ className?: string }>;
+    css?: CSS;
+}
+
+export const Icon = forwardRef<HTMLSpanElement, IconProps>(function Icon({ icon: IconComponent, css, ...rest }, ref) {
+    return (
+        <Styled.IconBox css={css} ref={ref} {...rest}>
```
Comment (adamtretera) [inline template/src/modules/ui/components/Icons/Icon/Icon.tsx:14]: You are not using the radix icon delete it or add it to this component

File: template/src/pages/_app.tsx [hunk 6]
```diff
@@ -1,25 +1,28 @@
...
     return (
-        <Fela renderer={renderer}>
-            <Intl>
-                <Component {...pageProps} />
-            </Intl>
-        </Fela>
+        <>
+                <Intl>
+                    <Component {...pageProps} />
+                    <Analytics />
```
Comment (adamtretera) [inline template/src/pages/_app.tsx]: I would also remove the Analytics as we are not using any soltion for the consent of tracking.

File: template/src/styles/globals.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    'html, body': {
+        fontFamily: '-apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif',
+    },
+
+    a: {
+        color: 'inherit',
+        textDecoration: 'none',
+    },
+
+    '*': {
```
Comment (adamtretera) [inline template/src/styles/globals.ts]:
```
*, *::before, *::after {
  box-sizing: border-box;
}
```

Maybe even add
```
img, picture, video, canvas, svg {
  display: block;
  max-width: 100%;
}
```

File: template/src/styles/theme/index.ts [hunk 8]
```diff
@@ -1,20 +1,39 @@
...
+        sizes: {
+            icon: '24px',
+            buttonIcon: '38px',
+            homepageContentWidth: '1282px',
+        },
+        space: {
+            containerPadding: '1.5rem',
+        },
+        zIndices: {
+            mobileNavigation: 10,
```
Comment (adamtretera) [inline template/src/styles/theme/index.ts]: I would just keep some basic font a color. smth. PrimaryColor 
We can delete the rest ?

File: template/src/styles/theme/index.ts [hunk 9]
```diff
@@ -1,20 +1,39 @@
...
+    },
+    media: {
+        tabletMax: '(max-width: 767px)',
+        tablet: '(min-width: 768px)',
+        tabletSmall: '(min-width: 620px)',
+        desktop: '(min-width: 1024px)',
+        desktopLarge: '(min-width: 1440px)',
+        headerBreakpoint: '(min-width: 1206px)',
+        maxHeaderBreakpoint: '(max-width: 1206px)',
+        headerChangeLayout: '(min-width: 900px)',
```
Comment (adamtretera) [inline template/src/styles/theme/index.ts]: Maybe just keep the mobile first breakpoints (so the defualt is mobile first) ???

File: template/src/modules/ui/components/Icons/Icon/Icon.styles.ts [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+        mobileInline: {
+            true: {
+                '@tabletMax': {
+                    display: 'inline-block',
+
+                    marginLeft: '0.25rem',
+                    transform: 'translateY(16%)',
+                },
+            },
+        },
```
Comment (adamtretera) [inline template/src/modules/ui/components/Icons/Icon/Icon.styles.ts]: This seems project specific ?

File: template/src/modules/ui/components/Icons/Icon/Icon.styles.ts [hunk 11]
```diff
@@ -0,0 +1,28 @@
+import { styled } from 'styles/theme';
+
+export const IconBox = styled('span', {
+    width: '$icon',
+    height: '$icon',
```
Comment (adamtretera) [inline template/src/modules/ui/components/Icons/Icon/Icon.styles.ts]: you could use the size util you specifed in the stithches config

File: template/src/modules/ui/components/Heading/Heading.styles.ts [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+
+export const Heading = styled('h1', {
+    fontFamily: '$museo',
+    fontWeight: 600,
+    margin: 0,
+
+    variants: {
+        level: {
+            1: {
+                fontSize: '2.8125rem',
```
Comment (adamtretera) [inline template/src/modules/ui/components/Heading/Heading.styles.ts]: I woudl keep the font sizes in the Stitchess config ?

Comment (adamtretera) [reply template/src/modules/ui/components/Heading/Heading.styles.ts]: https://github.com/radix-ui/design-system/blob/master/components/Text.tsx Maybe look how radix team does it, it seems like a prettty good approach

File: template/src/stories/Card.stories.tsx [hunk 13]
```diff
@@ -1,7 +1,6 @@
-import React from 'react';
 import type { ComponentMeta, ComponentStory } from '@storybook/react';
 
-import { Card } from './Card';
+import { Card } from 'modules/ui';
 
 export default {
     title: 'UI/Card',
```
Comment (adamtretera) [inline template/src/stories/Card.stories.tsx]: I would keep the stories alongside the component:
Its recomeded to do that: https://storybook.js.org/docs/react/writing-stories/introduction

File: template/src/stories/Heading.stories.tsx
Comment (adamtretera) [inline template/src/stories/Heading.stories.tsx:1]: Same here.

File: template/package.json (hunk 4 above)
Comment (jancama2) [reply template/package.json]: Agreed, I would keep react, jotai, react-query, stiches as the core part of our project. The rest is project dependent.  If we keep so many dependencies, we have to keep them up to date.

File: template/package.json (hunk 3 above)
Comment (jancama2) [reply template/package.json]: Agreed, the main "pro" of Radix is the ability to install only what you need; why to preinstall, then?

File: template/src/stories/Card.stories.tsx (hunk 13 above)
Comment (jancama2) [reply template/src/stories/Card.stories.tsx]: Agreed

File: template/src/styles/theme/index.ts [hunk 14]
```diff
@@ -1,20 +1,39 @@
...
-
-export type ThemeProps = { theme: Theme };
-
-export type TRuleWithTheme<Props = {}> = TRule<ThemeProps & Props>;
-
-export type RulesExtend<TExtandalbeRules, TProps = {}> = Partial<
-  Record<keyof TExtandalbeRules, TRuleWithTheme<TProps>>
->;
+export const { styled, css, globalCss, keyframes, getCssText, theme, createTheme, config } = createStitches({
+    theme: {
```
Comment (jancama2) [inline template/src/styles/theme/index.ts:4]: A lot of things here are project specific, I would delete it, you would do it when bootstraping a new project anyways.

File: template/src/styles/theme/index.ts (hunk 9 above)
Comment (jancama2) [reply template/src/styles/theme/index.ts]: Agreed and for me this seems super project specific, I would keep only 1 breakpoint for desktop and use the mobile-first approach.

**Review by jancama2 — CHANGES_REQUESTED** [review]
Generally, there is a lot of project specific things:
- deps - I would keep only the core thins - next, react, stitches, jotai, react-query; remove the rest
- components - keep only one component as an example, for example the card, and present all the options using that component, move story to the folder with the component, it is a good practices to have it alongside
- stitches theme - no need for so many predefined props as you will likely remove them for every new project, keep minimal breakpoints showing the mobile-first approach and that is just final

One more thing - I would add some basic test for the same component you will use for the stories example, add `@testing-library/react` and write a simple test as and example of component unit testing.

File: template/next.config.js (hunk 1 above)
Comment (baierjak) [reply template/next.config.js]: agreed, but I'm always wondering why images are not loaded :(

File: template/package.json (hunk 3 above)
Comment (baierjak) [reply template/package.json]: yes, definitely 👍

**Comment by baierjak** [comment]
> Generally, there is a lot of project specific things:
> 
> * deps - I would keep only the core thins - next, react, stitches, jotai, react-query; remove the rest
> * components - keep only one component as an example, for example the card, and present all the options using that component, move story to the folder with the component, it is a good practices to have it alongside
> * stitches theme - no need for so many predefined props as you will likely remove them for every new project, keep minimal breakpoints showing the mobile-first approach and that is just final
> 
> One more thing - I would add some basic test for the same component you will use for the stories example, add `@testing-library/react` and write a simple test as and example of component unit testing.

definitely, I've come from the new project where I have set everything up, and created this as the first draft, so thats the reason there are lot of project specific stuff left. Thanks for your points I will go through all of that and make it more clear 👍

File: template/package.json (hunk 4 above)
Comment (baierjak) [reply template/package.json]: agreed, I will removed all react query and jotai to have it clear now and I will add the minimalistic skelet for tanstack-query usage afterwards rather in the separate PR

File: template/src/pages/_app.tsx (hunk 6 above)
Comment (baierjak) [reply template/src/pages/_app.tsx]: or we can add some conditional to use it only on dev

File: template/src/styles/globals.ts (hunk 7 above)
Comment (baierjak) [reply template/src/styles/globals.ts]: ok, agree

File: template/src/styles/theme/index.ts (hunk 8 above)
Comment (baierjak) [reply template/src/styles/theme/index.ts]: ok no, problem, I just wanted to have it here for the easiest use

File: template/src/styles/theme/index.ts (hunk 9 above)
Comment (baierjak) [reply template/src/styles/theme/index.ts]: ok, but I suggest to leave here tablet, desktop, and larger desktop according to the sizes from our design team

```
tablet: '(min-width: 768px)',
desktop: '(min-width: 1024px)',
desktopLarge: '(min-width: 1440px)',
```

File: template/src/modules/ui/components/Icons/Icon/Icon.styles.ts (hunk 10 above)
Comment (baierjak) [reply template/src/modules/ui/components/Icons/Icon/Icon.styles.ts]: agree

File: template/src/modules/ui/components/Icons/Icon/Icon.styles.ts (hunk 11 above)
Comment (baierjak) [reply template/src/modules/ui/components/Icons/Icon/Icon.styles.ts]: good idea 👍

File: template/src/stories/Card.stories.tsx (hunk 13 above)
Comment (baierjak) [reply template/src/stories/Card.stories.tsx]: ok, makes sense

File: template/src/styles/theme/index.ts (hunk 14 above)
Comment (baierjak) [reply template/src/styles/theme/index.ts:4]: oks 👍

**Comment by baierjak** [comment]
All should be fixed, thanks for reviewing this.

Also easy unit tests for UI components added and also Chromatic library for storybook snapshot testing is added. 🎉

**Review by adamtretera — APPROVED** [review]

File: template/package.json [hunk 15]
```diff
@@ -52,43 +60,41 @@
         "@types/react-dom": "18.0.6",
         "@typescript-eslint/eslint-plugin": "5.6.0",
         "@typescript-eslint/parser": "5.6.0",
+        "chromatic": "^6.17.2",
```
Comment (cermakjiri) [inline template/package.json]: remove 🙏

File: template/package.json [hunk 16]
```diff
@@ -19,26 +20,33 @@
...
-        "fela-plugin-extend": "^12.2.0",
-        "fela-plugin-fallback-value": "^12.2.0",
-        "fela-plugin-prefixer": "^12.2.0",
-        "fela-plugin-validator": "^12.2.0",
-        "fela-tools": "^12.2.0",
+        "@stitches/react": "^1.2.8",
+        "@svgr/cli": "^6.5.1",
         "lodash": "^4.17.21",
-        "next": "12.2.2",
+        "next": "^13.2.4",
```
Comment (cermakjiri) [inline template/package.json]: upgrade to v13.4 without app folder structure 🙏

File: template/build-storybook.log
Comment (cermakjiri) [inline template/build-storybook.log:1]: add this to gitignore

File: template/lokse.config.ts [hunk 17]
```diff
@@ -2,10 +2,11 @@ import type { LokseConfig } from 'lokse';
 import { Languages } from './src/modules/intl/config';
 
 const config: LokseConfig = {
-    sheetId: '1q8MSZGQqsWzo6zdme3ChwuWF7EdXMAyeZWSrgwOp-YQ',
+    sheetId: '',
     dir: 'src/translations',
     languages: Object.values(Languages),
     column: 'key_web',
+    // sheets: [],
```
Comment (cermakjiri) [inline template/lokse.config.ts]: remove commented code

File: template/package.json (hunk 16 above)
Comment (cermakjiri) [reply template/package.json]: ~~with app folder structure~~ gonna be done later

File: template/package.json [hunk 18]
```diff
@@ -19,26 +20,33 @@
...
-        "configuru": "^0.6.2",
-        "fela": "^12.2.0",
-        "fela-monolithic": "^12.2.0",
-        "fela-plugin-extend": "^12.2.0",
-        "fela-plugin-fallback-value": "^12.2.0",
-        "fela-plugin-prefixer": "^12.2.0",
-        "fela-plugin-validator": "^12.2.0",
-        "fela-tools": "^12.2.0",
+        "@stitches/react": "^1.2.8",
+        "@svgr/cli": "^6.5.1",
```
Comment (cermakjiri) [inline template/package.json]: remove this, use rather `npx` (it doesn't have to be installed in CI during build)

File: template/package.json [hunk 19]
```diff
@@ -19,26 +20,33 @@
...
+        "@babel/core": "^7.20.12",
+        "@storybook/addon-actions": "^6.5.16",
+        "@storybook/addon-essentials": "^6.5.16",
+        "@storybook/addon-interactions": "^6.5.16",
+        "@storybook/addon-links": "^6.5.16",
+        "@storybook/builder-webpack5": "^6.5.16",
+        "@storybook/manager-webpack5": "^6.5.16",
+        "@storybook/react": "^6.5.16",
+        "@storybook/testing-library": "^0.1.0",
+        "babel-loader": "^8.3.0"
```
Comment (cermakjiri) [inline template/package.json]: Pls remove these and use `yarn install --optional` command 
```
   "storybook-addon-next": "^1.8.0",
        "tsconfig-paths-webpack-plugin": "^4.0.0",
        "@babel/core": "^7.20.12",
        "@storybook/addon-actions": "^6.5.16",
        "@storybook/addon-essentials": "^6.5.16",
        "@storybook/addon-interactions": "^6.5.16",
        "@storybook/addon-links": "^6.5.16",
        "@storybook/builder-webpack5": "^6.5.16",
        "@storybook/manager-webpack5": "^6.5.16",
        "@storybook/react": "^6.5.16",
        "@storybook/testing-library": "^0.1.0",
        "babel-loader": "^8.3.0"
```

File: template/package.json [hunk 20]
```diff
@@ -6,6 +6,7 @@
         "node": "16"
     },
     "scripts": {
+        "vercel:dev": "vercel env pull && vercel dev",
```
Comment (cermakjiri) [inline template/package.json]: rather rename to `dev:vercel`

**Review by cermakjiri — CHANGES_REQUESTED** [review]

File: template/package.json (hunk 18 above)
Comment (baierjak) [reply template/package.json]: unfortunatelly it is not working 

but it is moved to devDeps

image

Comment (cermakjiri) [reply template/package.json]: yes, because the command should be `npx @svgr/cli ...`

**Review by adamtretera — APPROVED** [review]

**Review by adamtretera — APPROVED** [review]

**Review by cermakjiri — APPROVED** [review]

File: template/.storybook/preview.jsx [hunk 21]
```diff
@@ -1,15 +1,10 @@
-import { Fela } from '../src/modules/fela';
-
-import '../src/styles/globals.css';
 
 export const parameters = {
     actions: { argTypesRegex: '^on[A-Z].*' },
 };
 
 export const decorators = [
```
Comment (jancama2) [inline template/.storybook/preview.jsx:6]: These decorators can be deleted, they do nothing in this case.

File: template/.gitlab-ci.yml [hunk 22]
```diff
@@ -1,8 +1,36 @@
...
-    - project: Ackee/gitlab-ci-templates
-      ref: nextjs-cloudrun/v1.0.1
-      file: nextjs-cloudrun/pipeline.yml
+default:
+  image: node:16.16.0
 
 variables:
-    NODE_BASE_IMAGE: 'node:16.19'
-    DEPLOYMENT_TYPE: 'cloudrun'
+  ACKEE_VERCEL_API_TOKEN: ""
```
Comment (jancama2) [inline template/.gitlab-ci.yml:5]: This should rather be in vault, not in a project, but I guess we can fine-tune the pipeline later.

File: template/src/pages/_app.tsx [hunk 23]
```diff
@@ -1,25 +1,25 @@
...
-    renderer: IRenderer;
 }
 
-export default function App({ Component, pageProps, renderer }: ExtendedAppProps) {
+function App({ Component, pageProps }: ExtendedAppProps) {
+    globalStyles();
+
     return (
-        <Fela renderer={renderer}>
+        <>
```
Comment (jancama2) [inline template/src/pages/_app.tsx:17]: Unnecessary fragment

**Review by jancama2 — APPROVED** [review]

**Pull request merged**